doc/design/fully_static_graph.md

```diff
@@ -0,0 +1,122 @@
+# Design Doc: Fully Static Graph
+
+## Abstract
+
+We propose the *fully static graph* rule: training and inference must
+be fully specified by the static graph. This means training and
+inference should be able to run solely on the cpp core (no Python
+involved), everything should be implemented as an OP.
+
+The user can still use Python to achieve the same result for
+convenience when experimenting locally, but the distributed training
+will not support Python.
```

Comment:
Does this mean that users cannot program distributed PaddlePaddle jobs in Python? I am afraid that most users might not agree.

Reply:
Thanks! I have modified the design doc, now the user can use Python.